composer-dependency-analyser.php

```diff
@@ -0,0 +1,76 @@
+<?php declare(strict_types = 1);
+
+use LastDragon_ru\LaraASP\Dev\App\Example;
+use ShipMonk\ComposerDependencyAnalyser\CliOptions;
+use ShipMonk\ComposerDependencyAnalyser\ComposerJson;
+use ShipMonk\ComposerDependencyAnalyser\Config\Configuration;
+use ShipMonk\ComposerDependencyAnalyser\Config\ErrorType;
+use ShipMonk\ComposerDependencyAnalyser\Initializer;
+use ShipMonk\ComposerDependencyAnalyser\Path;
+
+// Assertions
+assert(isset($this) && $this instanceof Initializer);
+assert(isset($options) && $options instanceof CliOptions);
+assert(isset($composerJson) && $composerJson instanceof ComposerJson);
```

Comment:
All this wont be available after https://github.com/shipmonk-rnd/composer-dependency-analyser/pull/102. Instead, you can:

1. get `$cwd` just by `getcwd()`
2. get the given path to `composer.json` by `getopt('', ['composer-json:'])['composer-json']`
3. parse autoload paths the same way you do for `exclude-from-classmap`
   - I'll think if I cannot add better support to override dev paths in this config so that you would not need to do `disableComposerAutoloadPathScan` (and re-adding those by `addPathToScan`).

Reply:
Basically any class or method not mentioned in readme (which is just `Configuration` and `ErrorType`) is not considered public API (as CDA is mainly a CI tool) and may change in future, so it is better to avoid using those.

Reply:
Yep, agree. But I will merge it as is (because need it for #117) and update a bit later.